core/src/test/java/com/linecorp/armeria/common/HttpResponseBuilderTest.java

```diff
@@ -0,0 +1,285 @@
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.contentUtf8()).isEqualTo("Armeriaはいろんな使い方がアルメリア");
+        assertThat(aggregatedRes.contentType()).isNotNull();
+        assertThat(aggregatedRes.contentType()).isEqualTo(MediaType.PLAIN_TEXT_UTF_8);
+    }
+
+    @Test
+    void buildWithContentTypeAndPlainFormat() {
+        final HttpResponse res = HttpResponse.builder()
+                                             .ok()
+                                             .content(MediaType.PLAIN_TEXT_UTF_8,
+                                                      "%sはいろんな使い方が%s",
+                                                      "Armeria", "アルメリア")
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.contentUtf8()).isEqualTo("Armeriaはいろんな使い方がアルメリア");
+        assertThat(aggregatedRes.contentType()).isNotNull();
+        assertThat(aggregatedRes.contentType()).isEqualTo(MediaType.PLAIN_TEXT_UTF_8);
+    }
+
+    @Test
+    void buildWithByteContent() {
+        final HttpResponse res = HttpResponse.builder()
+                                             .ok()
+                                             .content(MediaType.PLAIN_TEXT_UTF_8,
+                                                      "Armeriaはいろんな使い方がアルメリア"
+                                                              .getBytes(StandardCharsets.UTF_8))
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.contentUtf8()).isEqualTo("Armeriaはいろんな使い方がアルメリア");
+        assertThat(aggregatedRes.contentType()).isNotNull();
+        assertThat(aggregatedRes.contentType()).isEqualTo(MediaType.PLAIN_TEXT_UTF_8);
+    }
+
+    @Test
+    void buildWithPublisherContent() {
+        final HttpResponse res = HttpResponse.builder()
+                                             .ok()
+                                             .content(MediaType.PLAIN_TEXT_UTF_8,
+                                                      StreamMessage.of(
+                                                              HttpData.ofUtf8(
+                                                                      "Armeriaはいろんな使い方がアルメリア"
+                                                              )
+                                                      ))
+                                             .build();
+        final AggregatedHttpResponse aggregatedRes = res.aggregate().join();
+        assertThat(aggregatedRes.status()).isEqualTo(HttpStatus.OK);
+        assertThat(aggregatedRes.contentUtf8()).isEqualTo("Armeriaはいろんな使い方がアルメリア");
+        assertThat(aggregatedRes.contentType()).isNotNull();
```

Comment:
We can remove all these checks because we are checking the contentType anyway. 😄